services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -767,6 +769,99 @@ async def remove_tag(
                     project["tags"].remove(tag_id)
                 return convert_to_schema_names(project, user_email)
 
+    #
+    # Project Comments
+    #
+
+    async def create_project_comment(
+        self, project_uuid: ProjectID, user_id: UserID, content: str
+    ) -> CommentID:
+        async with self.engine.acquire() as conn:
+            project_comment_id: ResultProxy = await conn.execute(
+                projects_comments.insert()
+                .values(
+                    project_uuid=project_uuid,
+                    user_id=user_id,
+                    content=content,
+                    updated_at=func.now(),
+                )
+                .returning(projects_comments.c.comment_id)
+            )
+            result: tuple[PositiveInt] = await project_comment_id.first()
+            return parse_obj_as(CommentID, result[0])
+
+    async def list_project_comments(
+        self,
+        project_uuid: ProjectID,
+        offset: int | None = 0,
+        limit: int | None = None,
+    ) -> list[ProjectsCommentsDB]:
+        result = []
+        async with self.engine.acquire() as conn:
+            project_comment_result: ResultProxy = await conn.execute(
+                projects_comments.select()
+                .where(projects_comments.c.project_uuid == f"{project_uuid}")
+                .order_by(projects_comments.c.created_at.asc())
```

Comment:
Question: if you drop the ordering here, will the result be the same?

Reply:
I do not agree with this good practice. 
You force the DB to sort. It will still do the operation (even though fast). When you have lots of entries it will still need to scan the entire dataset and sort it. 
I was asking if the default sorting would be the same as ascending on created_at. Especially since, once again there is no index here. If you want to keep the sorting, sort on the primary key, which is auto incremented and provides the same result.

Reply:
We can discuss more in person tomorrow, but:
1. I am filtering by project_uuid, where I added the index as you recommended (meaning that you do not sort a whole dataset but a really small subset which is negligible 
2. Adding indexes should be done with caution where it really makes sense, each index increases the size of the database
3. Currently in the magnitude of data we are working with we do not need to preoptimize. For the sake of argument, if there wouldn't be an index on uuid field, it would still query really really fast. We can do some tests when we would start to feel the benefits of the index, I guess it would be when we would have more than 1mil rows.